client/blocks/credit-card-form/helpers.js

```diff
@@ -98,17 +99,27 @@ async function updateCreditCard( {
 	}
 
 	if ( purchase && siteSlug && isRenewable( purchase ) ) {
-		const noticeMessage = translate(
-			'Your credit card details were successfully updated, but your subscription has not been renewed yet.'
-		);
-		const noticeOptions = {
-			button: translate( 'Renew Now' ),
-			onClick: function( event, closeFunction ) {
-				handleRenewNowClick( purchase, siteSlug );
-				closeFunction();
-			},
-			persistent: true,
-		};
+		const daysAwayFromRenewal = moment( purchase.renewDate ).diff( moment(), 'days' );
+		let noticeMessage = '';
+		let noticeOptions = {};
+		if ( daysAwayFromRenewal > 90 ) {
```

Comment:
Should we maybe use `shouldAddPaymentSourceInsteadOfRenewingNow()` here instead, to be consistent with how other parts of the codebase make these kinds of decisions?

That's ~3 months away from _expiration_ (not the same as ~3 months away from renewal used here), but I think that's OK.  I also checked that using the expiration date to calculate the 3-month cutoff seems to be consistent with how we used to distinguish these cases server-side, back when we didn't allow early manual renewals at all (D19946-code).

Reply:
I like being consistent. Thanks for pointing this out, I didn't see it previously or I would've tried to use it. I concur with what you've found on D19946-code and thanks for digging that up. I've made this change, tested it and it appears to work properly still.